Add query migration handler

andresmgot opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana-plugin-sdk-go/blob/master/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

4. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the master branch.

-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

<!--

* Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer**:

andresmgot committed:
Add query migration handler

ryantxu commented:
hymm -- why not use [AdmissionControl](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/proto/backend.proto#L318)?  If it were urgent to get this out the door tomorrow, i could understand -- but given that it is not, it seems more valuable to exercise the k8s first approach.

Perhaps some go interfaces for this -- that are then backed by the existing admission controler protobuf?

andresmgot committed:
remove migrate grpc server

andresmgot committed:
remove unnecessary changes

andresmgot committed:
remove unnecesasry app changes

andresmgot commented:
Okay, I just moved the logic to run migrations here in the SDK (as suggested [here](https://github.com/grafana/grafana/pull/90334#discussion_r1675504958)), so this is now simplified 👍 

We will still need a way to call _only_ the migration logic from Grafana but it's not necessary for this PR.

andresmgot committed:
merge main
